app/controllers/articles_controller.rb

```diff
@@ -30,23 +29,36 @@ def edit
 
   def update
     @article = Article.find(params[:id])
-
-    if @article.update(article_params)
-      redirect_to @article
+    if @article.user_id == current_user.id
+      if @article.update(article_params)
+        redirect_to @article
+      else
+        render :edit
+      end
     else
-      render :edit
+      redirect_to @article
     end
   end
 
   def destroy
     @article = Article.find(params[:id])
+    if @article.user_id == current_user.id
     @article.destroy
-
-    redirect_to root_path
+    redirect_to articles_path
+    else
+      redirect_to @article
+    end
   end
 
   private
-    def article_params
-      params.require(:article).permit(:title, :body, :status)
+
+  def article_params
+    params.require(:article).permit(:title, :body, :status)
```

Comment:
@idprietop what **require** and **permit** does?

Reply:
requires an instance of an article and permit to receive the parameters: title, body and status.

Reply:
@idprietop thanks!